compiler/executable/function.rs

```diff
@@ -46,11 +46,17 @@ pub enum ExecutableReturn {
     Reduce(Vec<ReduceInstruction<VariablePosition>>),
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum FunctionTablingType {
+    Tabled,
+    Untabled,
+}
+
 pub(crate) fn compile_function(
     statistics: &Statistics,
     schema_functions: &ExecutableFunctionRegistry, // Can't have preamble in them when you're compiling functions
     function: AnnotatedFunction,
-    is_tabled: bool,
+    is_tabled: FunctionTablingType,
```

Comment:
Feels like `is_tabled` should be called differently now, t oo

Reply:
Good point. This is actually a boolean. I might shift to `pub struct IsFunctionTabled(bool)`

Reply:
Choosing the right approach might actually be a good question for @dmitrii-ubskii. In C++, we used to put comments near the boolean params passed into functions and force the vertical alignment of params (blocking some of the auto-formatters to rearrange it):
```
person.yawn(
    duration,
    false // isLoud
);
```
I don't know what's the common approach in Rust.

Reply:
Creating structs / enums / whatever for every boolean we want to visually annotate can be annoying.